lib/msf/core/post/file.rb

```diff
@@ -638,6 +642,153 @@ def copy_file(src_file, dst_file)
+      #
+      # Group Name                                 Type             SID                                            Attributes
+      #
+      # ========================================== ================ ============================================== ==================================================
+      # Everyone                                   Well-known group S-1-1-0                                        Mandatory group, Enabled by default, Enabled group
+      # BUILTIN\Users                              Alias            S-1-5-32-545                                   Mandatory group, Enabled by default, Enabled group
+      # BUILTIN\Performance Log Users              Alias            S-1-5-32-559                                   Mandatory group, Enabled by default, Enabled group
+      # BUILTIN\Administrators                     Alias            S-1-5-32-544                                   Group used for deny only
+      # NT AUTHORITY\INTERACTIVE                   Well-known group S-1-5-4                                        Mandatory group, Enabled by default, Enabled group
+      # CONSOLE LOGON                              Well-known group S-1-2-1                                        Mandatory group, Enabled by default, Enabled group
+      # NT AUTHORITY\Authenticated Users           Well-known group S-1-5-11                                       Mandatory group, Enabled by default, Enabled group
+      # NT AUTHORITY\This Organization             Well-known group S-1-5-15                                       Mandatory group, Enabled by default, Enabled group
+      # LOCAL                                      Well-known group S-1-2-0                                        Mandatory group, Enabled by default, Enabled group
+      #                                            Unknown SID type S-1-5-21-3070936213-306261907-1348773959-36273 Mandatory group, Enabled by default, Enabled group
+      #                                            Unknown SID type S-1-5-21-3070936213-306261907-1348773959-45105 Mandatory group, Enabled by default, Enabled group
+      # **** CUT FOR BREVITY ***
+      # Authentication authority asserted identity Well-known group S-1-18-1                                       Mandatory group, Enabled by default, Enabled group
+      # Mandatory Label\Medium Mandatory Level     Label            S-1-16-8192
+      #
+      # C:\Users>
+
+      cuser_groups_array = win_find_current_user_groups
+
+      # Yes its nested array hell, but until I have a
+      # better approach it is what it is.
+      #
+      # Loop around and for each group our user is a part of,
+      # see if that group has the desired permissions on the specified path.
+      for group in cuser_groups_array
+        for entry in permissions_file_array
+          next unless entry[0].include?(group)
+
+          for perm in perms
+            return true if entry[1].include?(perm)
+          end
+        end
+      end
```

Comment:
```suggestion
cuser_groups_array.each do |group|
  perms.each do |perm|
    file_icacls_groups[group].include? perm
      return true
    end
 end
```

Reply:
@jmartin-r7 Where is `file_icacls_group` coming in here? Not seeing where that is retrieved from.

Reply:
Suggested change implemented in https://github.com/rapid7/metasploit-framework/pull/15987/commits/21be7a0ee8a0de1cc3ab01a8d117819122a89120